mldsa/poly.h

```diff
@@ -70,10 +70,12 @@ void poly_add(poly *r, const poly *b)
 __contract__(
   requires(memory_no_alias(b, sizeof(poly)))
   requires(memory_no_alias(r, sizeof(poly)))
-  requires(forall(k0, 0, MLDSA_N, (int64_t) r->coeffs[k0] + b->coeffs[k0] <= INT32_MAX))
+  requires(forall(k0, 0, MLDSA_N, (int64_t) r->coeffs[k0] + b->coeffs[k0] < REDUCE32_DOMAIN_MAX))
```

Comment:
Do we really need that pre-condition here (and the corresponding post-condition)?
The post-condition is already a functional spec of addition. So this reasoning should be possible on a higher level?